snapcraft/meta/manifest.py

```diff
@@ -40,41 +40,35 @@ class Manifest(YamlModel):
     version: str
     summary: str
     description: str
-    base: Optional[str]
+    base: str | None = None
     grade: str
     confinement: str
-    apps: Optional[Dict[str, Any]]
```

Comment:
In every file where I did `Optional[T]` -> `T | None = None` fixes, I updated the typing as well (`Dict` -> `dict`, etc).